# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/base/nsGlobalWindowInner.cpp
# Commit: 6ff3e88d604f
# Full Hash: 6ff3e88d604fcc7afd50208a784a1307acce173c
# Author: Tim Huang <tihuang@mozilla.com>
# Date: 2020-07-15 16:06:28
# Description:
#   Bug 1650655 - Don't get PermissionDelegateHandler if the document is not ready yet in nsGlobalWindowInner.cpp. r=baku
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D83609
# ==============================================================================

diff -r 4bdbe6d4d658 -r 6ff3e88d604f dom/base/nsGlobalWindowInner.cpp
--- a/dom/base/nsGlobalWindowInner.cpp	Fri Jul 10 13:40:08 2020 +0000
+++ b/dom/base/nsGlobalWindowInner.cpp	Wed Jul 15 10:04:40 2020 +0000
@@ -4986,6 +4986,10 @@
       UpdateAutoplayPermission();
     }
 
+    if (!mDoc) {
+      return NS_OK;
+    }
+
     RefPtr<PermissionDelegateHandler> permDelegateHandler =
         mDoc->GetPermissionDelegateHandler();
 
